### PR TITLE
Add convenience method hasAttibute() to SAMRecord

### PR DESCRIPTION
**Pull request by SHuang-Broad**
### Description

As described in title.

### Checklist

- [x] Code compiles correctly
- [x] New tests covering changes and new functionality
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

File: src/main/java/htsjdk/samtools/SAMRecord.java [hunk 1]
```diff
@@ -1352,16 +1356,22 @@ private void requireUnsigned(final String tag) {
     }
 
     /**
+     * @return SAMBinaryTagAndValue may be null if
+     *          1) {@link #mAttributes} is null or
+     *          2) this record does not have the requested attribute
+     */
+    private SAMBinaryTagAndValue hasAttribute(final short tag) {
```
Comment (lbergelson) [inline src/main/java/htsjdk/samtools/SAMRecord.java]: This seems like it's wrong.  It should return a boolean.

Comment (lbergelson) [reply src/main/java/htsjdk/samtools/SAMRecord.java]: Or it could be named something else.

File: src/main/java/htsjdk/samtools/SAMRecord.java [hunk 2]
```diff
@@ -1116,6 +1116,10 @@ public void setValidationStringency(final ValidationStringency validationStringe
         this.mValidationStringency = validationStringency;
     }
 
+    public boolean hasAttribute(final String tag) {
```
Comment (lbergelson) [inline src/main/java/htsjdk/samtools/SAMRecord.java]: needs javadoc

File: src/main/java/htsjdk/samtools/SAMRecord.java [hunk 3]
```diff
@@ -1116,6 +1116,10 @@ public void setValidationStringency(final ValidationStringency validationStringe
...
+        return hasAttribute(SAMTagUtil.getSingleton().makeBinaryTag(tag))!=null;
```
Comment (lbergelson) [inline src/main/java/htsjdk/samtools/SAMRecord.java]: shouldn't this just be something like `getAttribute != null?`

File: src/main/java/htsjdk/samtools/SAMRecord.java (hunk 2 above)
Comment (SHuang-Broad) [reply src/main/java/htsjdk/samtools/SAMRecord.java]: done. Thought the name is very obvious...

File: src/main/java/htsjdk/samtools/SAMRecord.java (hunk 3 above)
Comment (SHuang-Broad) [reply src/main/java/htsjdk/samtools/SAMRecord.java]: done

File: src/main/java/htsjdk/samtools/SAMRecord.java (hunk 1 above)
Comment (SHuang-Broad) [reply src/main/java/htsjdk/samtools/SAMRecord.java]: renamed

**Comment by SHuang-Broad** [comment]
@lbergelson updated, please take another look. Thanks!

File: src/main/java/htsjdk/samtools/SAMRecord.java [hunk 4]
```diff
@@ -1352,16 +1359,22 @@ private void requireUnsigned(final String tag) {
     }
 
     /**
+     * @return SAMBinaryTagAndValue may be null if
+     *          1) {@link #mAttributes} is null or
+     *          2) this record does not have the requested attribute
+     */
+    private SAMBinaryTagAndValue getAttributeMayBeNull(final short tag) {
```
Comment (lbergelson) [inline src/main/java/htsjdk/samtools/SAMRecord.java]: @SHuang-Broad Do we need this method at all?  I still don't really understand why we need to extract this out.

Comment (yfarjoun) [reply src/main/java/htsjdk/samtools/SAMRecord.java]: is `getNullableAttribute` more consistent with Java naming schemes?

Comment (lbergelson) [reply src/main/java/htsjdk/samtools/SAMRecord.java]: it's not a naming issue anymore, it's just that if there's only 1 use of this, I don't see the need to extract it at all.

**Comment by SHuang-Broad** [comment]
@lbergelson removed the refactoring. Please take another look. Thanks!

__EDIT__
not sure about what happened to the "15 files have unexpected coverage changes..."

**Comment by lbergelson** [comment]
@SHuang-Broad 👍

**Pull request merged**